src/cloud/components/Comments/CommentReactions.tsx

```diff
@@ -0,0 +1,167 @@
+import { Comment } from '../../interfaces/db/comments'
+import { SerializedUser } from '../../interfaces/db/user'
+import CommentEmoji from '../CommentEmoji'
+import React, { useCallback, useMemo } from 'react'
+import Button from '../../../design/components/atoms/Button'
+import { mdiPlus } from '@mdi/js'
+import { EmojiReactionData } from './CommentList'
+import styled from '../../../design/lib/styled'
+import EmojiPickHandler from './EmojiPickHandler'
+import { CommentReaction } from '../../interfaces/db/commentReaction'
+import { Emoji } from 'emoji-mart'
+
+export type CommentReactionsProps = {
+  comment: Comment
+  users: SerializedUser[]
+  addReaction: (comment: Comment, emoji: string) => Promise<any>
+  removeReaction: (comment: Comment, reactionId: string) => Promise<any>
+  user?: SerializedUser
+}
+
+const CommentReactions = ({
+  comment,
+  addReaction,
+  removeReaction,
+  users,
+  user,
+}: CommentReactionsProps) => {
+  const filteredReactions = useMemo(() => {
+    const emojiReduce = (reactions: CommentReaction[], key: string) =>
```

Comment:
Function name should be predicate. `mapEmoji`.
Don't use an arrow function for a helper method inside of method. Also please consider pulling out if it is possible.